src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUnary.cs

```diff
@@ -425,19 +426,26 @@ public static DateValue NumberToDate(IRContext irContext, NumberValue[] args)
             return new DateValue(irContext, date);
         }
 
-        public static DateTimeValue NumberToDateTime(EvalVisitor runner, EvalVisitorContext context, IRContext irContext, NumberValue[] args)
+        public static FormulaValue NumberToDateTime(EvalVisitor runner, EvalVisitorContext context, IRContext irContext, NumberValue[] args)
         {
             return NumberToDateTime(CreateFormattingInfo(runner), irContext, args[0]);
         }
 
-        public static DateTimeValue NumberToDateTime(FormattingInfo formatInfo, IRContext irContext, NumberValue value)
+        public static FormulaValue NumberToDateTime(FormattingInfo formatInfo, IRContext irContext, NumberValue value)
         {
             var n = value.Value;
-            var date = _epoch.AddDays(n);
```

Comment:
>       var date = _epoch.AddDays(n); [](http://example.com/codeflow?start=5&length=36)

This is where the failure is coming from (throws a ArgumentOutOfRangeException) 

We could just catch that and rethrow as CustomFunctionErrorException. That will get catch in EvalVisitor. 

That lets us keep the signature type safe (DateTimeValue instead of downgrading to FormulaValue); and means callers don't have to all check for failure.

Reply:
else if we wanted  DateTimeValue or ErrroValue, we could use DValue<DateTimeValue>